lib/Sema/CSApply.cpp

```diff
@@ -5827,9 +5827,11 @@ Expr *ExprRewriter::coerceCallArguments(Expr *arg, AnyFunctionType *funcType,
       // We already had a ParenExpr, so replace it's sub-expression.
       argParen->setSubExpr(newArgs[0]);
     } else {
+      bool isImplicit = arg->isImplicit();
       arg = new (ctx)
           ParenExpr(lParenLoc, newArgs[0], rParenLoc, hasTrailingClosure);
-      arg->setImplicit();
+      if (isImplicit)
+        arg->setImplicit();
```

Comment:
FYI `Expr::setImplicit` accepts an argument so I believe `arg->setImplicit(isImplicit);` should work.